### ISSUE TITLE
Resources.getResource() not working on modular project (openjdk 11)? 

### ISSUE THREAD

**Issue by GG-A** [google/guava#3868]
this is a **maven project**, and have one image in **resources directory**:

```
├─ src
   ├─ main
       ├─ java
       └─ resources  
          └─imgs
            └─logo.png
```

I use: 
```
        <dependency>
            <groupId>com.google.guava</groupId>
            <artifactId>guava</artifactId>
            <version>29.0-jre</version>
        </dependency>
```
### Code Snippet:
```
public class Test {
    public static void main(String[] args) {
        URL resource = Resources.getResource(Test.class, "/imgs/logo.png");
    }
|
```
Will get error, when I **add** `module-info.java`:
```
Exception in thread "main" java.lang.IllegalArgumentException: resource /imgs/logo.png relative to com.ctxx.Test not found.
	at com.google.common@29.0-jre/com.google.common.base.Preconditions.checkArgument(Preconditions.java:441)
	at com.google.common@29.0-jre/com.google.common.io.Resources.getResource(Resources.java:208)
	at OpenJFXAndOpenJDK/com.ctxx.Test.main(Test.java:15)
```
but when I **delete** `module-info.java` from project, it works well.

### Code Snippet2:
```
public class Test {
    public static void main(String[] args) {
        URL resource1 = Resources.getResource("imgs/logo.png");
        System.out.println(resource1);
    }
}

```
when I **add** `module-info.java` to project, get error:
```
Exception in thread "main" java.lang.IllegalArgumentException: resource imgs/logo.png not found.
	at com.google.common@29.0-jre/com.google.common.base.Preconditions.checkArgument(Preconditions.java:217)
	at com.google.common@29.0-jre/com.google.common.io.Resources.getResource(Resources.java:195)
	at OpenJFXAndOpenJDK/com.ctxx.Test.main(Test.java:14)
```
but when I **delete** `module-info.java` from project, it works well.

**Comment by netdpb** [comment]
I think that JPMS says that if a resource is in a module, then `ClassLoader.getResource()` will hide it from callers of that method that aren't in the module. So _maybe_ if you call `getClass().getClassLoader().getResource()` directly instead of `Resources.getResource()` it'll work. 

I'm not a JPMS expert, but putting `opens imgs to com.google.common;` into your `module-info.java` should tell JPMS that Guava is allowed to load resources (and do other reflection) on the `imgs` package in your module.

Otherwise you can put `opens imgs;` into `module-info.java` to make that packages open to reflection by every module.

I don't think this is a Guava-specific issue, so I'm going to close this bug. Please reopen if you think there's something else we can do here.

**Comment by cpovirk** [comment]
I suspect that using a different classloader isn't enough. I can't find the specific JPMS reference that I thought I had bookmarked, but I came across [A JPMS Migration Guide for the Impatient](https://dzone.com/articles/java-platform-module-systemjpms-migration-guide-fo), which suggests that [`Module.getResourceAsStream`](https://docs.oracle.com/javase/9/docs/api/java/lang/Module.html#getResourceAsStream-java.lang.String-) may be somewhat better than [`ClassLoader.getResourceAsStream`](https://docs.oracle.com/javase/9/docs/api/java/lang/ClassLoader.html#getResourceAsStream-java.lang.String-). (Even the `Module` version is sometimes not sufficient, but I suspect that it is for the submitter of this issue, since `imgs` is probably not the name of a Java package.) That doc also mentions a `jrt:` URL scheme, about which I know nothing.

I think we should have some issue open about `Resources.getResource` and JPMS, covering:

- documentation on it similar to that on `ClassLoader.getResourceAsStream`
- feature requests for additional methods like `Modules.resourceAsByteSource`

That could be this issue, one new issue, or multiple new issues. Opinions?
